src/common/tracer.h

```diff
@@ -0,0 +1,64 @@
+#ifndef TRACER_H_
+#define TRACER_H_
+
+#define SIGNED_RIGHT_SHIFT_IS 1
+#define ARITHMETIC_RIGHT_SHIFT 1
+
+#include <arpa/inet.h>
+#include <yaml-cpp/yaml.h>
+
+#include <jaegertracing/Tracer.h>
+
+//forward declaration of req_state defined /rgw/rgw_common.h
+struct req_state;
+
+typedef std::unique_ptr<opentracing::Span> Span;
+
+class Jager_Tracer{
+  public:
+    Jager_Tracer(){}
+    ~Jager_Tracer(){
+      if(this->tracer)
+        this->tracer->Close();
+    }
+
+    void init_tracer(const char* tracerName,const char* filePath){
+        auto yaml = YAML::LoadFile(filePath);
+        auto configuration = jaegertracing::Config::parse(yaml);
+        this->tracer = jaegertracing::Tracer::make(
+        tracerName,
+        configuration,
+        jaegertracing::logging::consoleLogger());
+        opentracing::Tracer::InitGlobal(
+        std::static_pointer_cast<opentracing::Tracer>(tracer));
+    }
+    inline void finish_tracer(){
```

Comment:
i might have forgotten the context, is this function of use(providing any abstraction), since it's essentially holding the same behaviour as `tracer->Close`

Reply:
it is used to close a tracer, you told to me make a function to close tracer

Reply:
if you don't have any other purpose for it, then it's removal sounds relevant.

Reply:
okay then I will remove it